src/main/java/com/ioet/bpm/skills/boundaries/CategoryController.java

```diff
@@ -0,0 +1,90 @@
+package com.ioet.bpm.skills.boundaries;
+
+import com.ioet.bpm.skills.domain.Category;
+import com.ioet.bpm.skills.repositories.CategoryRepository;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+
+    private final CategoryRepository categoryRepository;
+
+    public CategoryController(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @ApiOperation(value = "Return a list of all categories", response = Category.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Categories successfully returned")
+    })
+    @GetMapping(produces = "application/json")
+    public ResponseEntity<Iterable> getAllCategories() {
+        Iterable<Category> categories = this.categoryRepository.findAll();
+        return new ResponseEntity<>(categories, HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Return one category", response = Category.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Category successfully returned")
+    })
+    @GetMapping(path = "/{id}", produces = "application/json")
+    public ResponseEntity<Category> getCategory(@PathVariable(value = "id") String categoryId) {
+        Optional<Category> categoryFromDB = categoryRepository.findById(categoryId);
+        return categoryFromDB.map(
+                skill -> new ResponseEntity<>(categoryFromDB.get(), HttpStatus.OK))
```

Comment:
I know I'm not requested a reviewer, but shouldn't it say 
`category -> new ResponseEntity` ?